src/executor/graph_executor.ts

```diff
@@ -221,17 +224,37 @@ export class GraphExecutor {
             key => this.weightMap[key].forEach(tensor => tensor.dispose()));
   }
 
+  private checkInputShapeAndType(inputs: NamedTensorsMap) {
+    this.placeholders.forEach(node => {
+      const input = inputs[node.name][0];
+      if (node.params['shape'] && node.params['shape'].value) {
+        const shape = node.params['shape'].value as number[];
+        const match = shape.length === input.shape.length &&
+            input.shape.every(
+                (dim, index) => shape[index] === -1 || shape[index] === dim);
+        util.assert(
+            match, `${node.name} shape(${input.shape}) must be ${shape}`);
+      }
+      if (node.params['dtype'] && node.params['dtype'].value) {
+        util.assert(
+            input.dtype === node.params['dtype'].value as string,
+            `The dtypes of the ${node.name}(${input.dtype}) and` +
```

Comment:
This error message doesn't quite make sense gramatically:

perhaps:

The dtype of the input is `float32` but `int32` was expected'